Mathlib/NumberTheory/Rayleigh.lean

```diff
@@ -0,0 +1,190 @@
+    exact sub_nonneg.1 (Int.sub_floor_div_mul_nonneg (j + 1 : ℝ) h)
+
+end Beatty
+
+/-- Generalization of Rayleigh's theorem on Beatty sequences. Let `r` be a real number greater
+than 1, and `1/r + 1/s = 1`. Then the complement of `B_r` is `B'_s`. -/
+theorem rayleigh_compl {r s : ℝ} (hrs : r.IsConjugateExponent s) :
+    {beattySequence r k | k}ᶜ = {beattySequence' s k | k} := by
+  ext j
+  by_cases h₁ : j ∈ {beattySequence r k | k}
+  · by_cases h₂ : j ∈ {beattySequence' s k | k}
+    · exact (Set.not_disjoint_iff.2 ⟨j, h₁, h₂⟩ (Beatty.no_collision hrs)).elim
+    · simp only [Set.mem_compl_iff, h₁, h₂]
+  · by_cases h₂ : j ∈ {beattySequence' s k | k}
+    · simp only [Set.mem_compl_iff, h₁, h₂]
+    · have ⟨k, h₁₁, h₁₂⟩ := (Beatty.hit_or_miss hrs.pos).resolve_left h₁
+      have ⟨m, h₂₁, h₂₂⟩ := (Beatty.hit_or_miss' hrs.symm.pos).resolve_left h₂
+      exact (Beatty.no_anticollision hrs ⟨j, k, m, h₁₁, h₁₂, h₂₁, h₂₂⟩).elim
+
+/-- Generalization of Rayleigh's theorem on Beatty sequences. Let `r` be a real number greater
+than 1, and `1/r + 1/s = 1`. Then `B⁺_r` and `B⁺'_s` partition the positive integers. -/
+theorem rayleigh_pos {r s : ℝ} (hrs : r.IsConjugateExponent s) :
+    {beattySequence r k | k > 0} ∆ {beattySequence' s k | k > 0} = {n | 0 < n} := by
+  apply Set.eq_of_subset_of_subset
+  · intro j hj
+    rcases Set.mem_of_mem_of_subset hj Set.symmDiff_subset_union with ⟨k, hk, hjk⟩ | ⟨k, hk, hjk⟩
+    · rw [Set.mem_setOf_eq, ← hjk, beattySequence, Int.floor_pos]
+      exact one_le_mul_of_one_le_of_one_le (by norm_cast) hrs.one_lt.le
+    · rw [Set.mem_setOf_eq, ← hjk, beattySequence', ← Int.ceil_sub_one, Int.ceil_pos, sub_pos]
+      exact one_lt_mul_of_le_of_lt (by norm_cast) hrs.symm.one_lt
+  intro j (hj : 0 < j)
+  have hb₁ : ∀ s ≥ 0, j ∈ {beattySequence s k | k > 0} ↔ j ∈ {beattySequence s k | k} := by
+    intro _ hs
+    refine ⟨fun ⟨k, _, hk⟩ ↦ ⟨k, hk⟩, fun ⟨k, hk⟩ ↦ ⟨k, ?_, hk⟩⟩
+    rw [← hk, beattySequence, Int.floor_pos] at hj
+    have := pos_of_mul_pos_left (lt_of_lt_of_le zero_lt_one hj) hs
+    rwa [Int.cast_pos] at this
+  have hb₂ : ∀ s ≥ 0, j ∈ {beattySequence' s k | k > 0} ↔ j ∈ {beattySequence' s k | k} := by
+    intro _ hs
+    refine ⟨fun ⟨k, _, hk⟩ ↦ ⟨k, hk⟩, fun ⟨k, hk⟩ ↦ ⟨k, ?_, hk⟩⟩
+    rw [← hk, beattySequence', lt_sub_iff_add_lt, zero_add] at hj
+    have hj := Int.ceil_pos.1 (lt_trans zero_lt_one hj)
+    have := pos_of_mul_pos_left hj hs
+    rwa [Int.cast_pos] at this
+  rw [Set.mem_symmDiff, hb₁ _ hrs.nonneg, hb₂ _ hrs.symm.nonneg, ← rayleigh_compl hrs,
+    Set.not_mem_compl_iff, Set.mem_compl_iff, and_self, and_self]
+  exact or_not
+
+/-- Rayleigh's theorem on Beatty sequences. Let `r` be an irrational number greater than 1, and
+`1/r + 1/s = 1`. Then `B⁺_r` and `B⁺_s` partition the positive integers. -/
+theorem rayleigh_irr_pos {r s : ℝ} (hrs : r.IsConjugateExponent s) (hr : Irrational r) :
+    {beattySequence r k | k > 0} ∆ {beattySequence s k | k > 0} = {n | 0 < n} := by
```

Comment:
This proof sounds too complicated given that you should just say `Irrational r → beattySequence' r = beattySequence r` (which could be a simp lemma, btw) and `r.IsConjugateExponent s → (Irrational r ↔ Irrational s)`.

Reply:
Proof of `rayleigh_irr_pos` has been shorted to 1 line.